eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/NoEventCriteria.java

```diff
@@ -51,7 +52,12 @@ public Set<String> types() {
     }
 
     @Override
-    public Set<Index> indices() {
+    public Set<Tag> tags() {
         return Set.of();
     }
+
+    @Override
+    public boolean matchingTags(@Nonnull Set<Tag> tags) {
+        return true;
```

Comment:
I'm not sure if `NotEventCriteria` name is accurate. 
If we want to return all events (as I understand from the implementation) maybe `AnyEventCriteria` will be better aligned with the behaviour?

Reply:
From the perspective of an append condition, there might not be any criteria when appending. This can occur when publishing out of the scope of a model/aggregate, for example.
But, when reading, calling it "no criteria" is...well, off, as you essentially allow any event. 

From appending, however, this should default to not care for any conditions. That's not a `Set<Tag>` thing, though, but a `consistencyMarker` pointer. @abuijze already correctly adjusted the default `consistencyMarker` for the `NoAppendCondition` to `Long.MAX_VALUE` in [this PR](https://github.com/AxonFramework/AxonFramework/pull/3216/files#diff-247b597354e05788fa2c8dfd9c27c1cd97619c51bbcb9ad3184dd9517d6c194f).

Thus, my change of the `matchingTags` method, your rename suggestion, and Allard's default `consistencyMarker` change, will make the `NoAppendCondition` and `NoEventCriteria` work as required.

Ow, lastly, I renamed to `AnyEvent` instead of `AnyEventCriteria`. Adding `Criteria` to the name felt slightly redundant. Let me know if you disagree!